app/models/task.rb

```diff
@@ -71,20 +68,23 @@ def retry_task(force=false)
     # setting this causes this post commit callback to send out the message
     self.process_task = true
 
-    task_log = task_logs.create(status: RETRYING, message: "retrying task: status: #{status}, force: #{force}", logged_at: Time.now)
+    task_log = task_logs.create(status: RETRYING, message: "retrying task: status: #{status}, force: #{force}", logged_at: Time.now, info: {})
     self.retrying!
     task_log
   end
 
   def handle_status_changes
-    return if previous_status.to_s == status.to_s
+    logger.debug "task: publish_messages: status: #{status}"
+
+    return unless previous_changes['status']
 
+    previous_status = previous_changes['status'].try(:first)
     logger.debug "task: publish_messages: #{previous_status} != #{status}"
 
     if process_task && created?
       logger.debug "task: publish_messages: insert created log"
       # log create on after_commit
-      task_logs.create!(status: CREATED, message: 'created message.', logged_at: Time.now)
+      task_logs.create!(status: CREATED, message: 'created message.', logged_at: Time.now, info: {})
```

Comment:
Line is too long. [100/80]<br>Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.